### PR TITLE
fix python watch dog not work

### PR DESCRIPTION
**Pull request by casionone**
### Related issues/PRs

Related issues: #4294 

### Brief change log

optimize pyspark watch dog

### CODE REVIEW

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkPythonExecutor.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ class SparkPythonExecutor(val sparkEngineSession: SparkEngineSession, val id: In
         // process.destroy will kills the subprocess,not need to force kill with -9,
         // kill -9 may cause resources not to be released
         pid.foreach(p => {
-          logger.info(s"skip Try to kill Pyspark process with: [kill -9 ${p}]")
+          logger.info(s"skip Try to kill Pyspark process with: [kill -15 ${p}]")
           Utils.exec(Array("kill", "-9", p), 3000L)
```
Comment (jackxu2011) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkPythonExecutor.scala:130]: the log is kill -15, but the exec cmd is kill -9

File: linkis-engineconn-plugins/spark/src/main/resources/python/mix_pyspark.py [hunk 2]
```diff
@@ -209,7 +209,7 @@ def saveDFToCsv(df, path, hasheader=True,isOverwrite=False,option={}):
 
 ##add pyfiles
 try:
-    pyfile = sys.argv[5]
+    pyfile = sys.argv[6]
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/resources/python/mix_pyspark.py:212]: should args is 5

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkPythonExecutor.scala [hunk 3]
```diff
@@ -187,11 +194,24 @@ class SparkPythonExecutor(val sparkEngineSession: SparkEngineSession, val id: In
       .foreach(pythonClasspath ++= File.pathSeparator ++= _)
 
     val cmd = CommandLine.parse(pythonExec)
-    cmd.addArgument(createFakeShell(pythonScriptPath).getAbsolutePath, false)
+    if (SparkConfiguration.SPARK_PYTHON_TEST_MODE_ENABLE.getValue) {
+      val path = SparkConfiguration.SPARK_PYTHON_TEST_MODE_MIX__PYSHELL_PATH.getValue
+      logger.info(
+        s"wds.linkis.spark.python.test.mode.enable => true, will use wds.linkis.spark.python.mix.pyshell.path:${path}"
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkPythonExecutor.scala:200]: Should SparkConfiguration.SPARK_PYTHON_TEST_MODE_ENABLE.key

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**